packages/jest/src/generators/init/init.ts

```diff
@@ -100,10 +101,12 @@ function createJestConfig(tree: Tree, options: NormalizedSchema) {
       const isProjectConfig = jestTarget?.options?.jestConfig === rootJestPath;
       // if root project doesn't have jest target, there's nothing to migrate
       if (isProjectConfig) {
-        const jestAppConfig = `jest.config.app.${options.js ? 'js' : 'ts'}`;
+        const jestProjectConfig = `jest.config.${
+          rootProjectConfig.projectType === 'application' ? 'app' : 'lib'
+        }.${options.js ? 'js' : 'ts'}`;
```

Comment:
question (non-blocking): Is there a case where the root project isn't an application?

Reply:
Yes, when TS standalone is used the root project is a TS library.